app/controllers/proposal_controller.rb

```diff
@@ -47,39 +49,38 @@ def create
     @event = Event.new(event_params)
     @event.program = @program
 
+    # User which creates the proposal is both `submitter` and `speaker` of proposal
+    # by default.
+    # TODO: Allow submitter to add speakers to proposals
     @event.event_users.new(user: current_user,
                            event_role: 'submitter')
     @event.event_users.new(user: current_user,
                            event_role: 'speaker')
     authorize! :new, @event
 
-    unless @event.save
+    if @event.save
+      ahoy.track 'Event submission', title: 'New submission'
+      redirect_to conference_program_proposal_index_path(@conference.short_title), notice: 'Proposal was successfully submitted.'
+    else
       flash[:error] = "Could not submit proposal: #{@event.errors.full_messages.join(', ')}"
       render action: 'new'
-      return
     end
-
-    ahoy.track 'Event submission', title: 'New submission'
-
-    redirect_to conference_program_proposal_index_path(@conference.short_title), notice: 'Proposal was successfully submitted.'
   end
 
   def update
-    authorize! :update, @event
     @url = conference_program_proposal_path(@conference.short_title, params[:id])
 
-    if !@event.update(event_params)
+    if @event.update(event_params)
+      redirect_to(conference_program_proposal_index_path(conference_id: @conference.short_title),
```

Comment:
I have reverted this change. Should I revert other ([here](https://github.com/openSUSE/osem/pull/907/files#diff-8c13e9cfe1f5edf7d4923e633a08f790R141) and [here](https://github.com/openSUSE/osem/pull/907/files#diff-8c13e9cfe1f5edf7d4923e633a08f790R61)) similar changes as well? I had made the change cause following is the style we follow in rest of controllers:
```Ruby
if @object.save
  # do something
else 
  # do something else
end
```
instead of:
```Ruby
if !@object.save
   # do something
   return
end

# do something
```